cmd/monaco/deploy/deploy.go

```diff
@@ -242,3 +247,23 @@ func collectRequiresPlatformErrors(platformCoordinatesPerEnvironment Coordinates
 func platformEnvironment(e manifest.EnvironmentDefinition) bool {
 	return e.Auth.OAuth != nil
 }
+
+func validateAuthenticationWithProjectConfigs(projects []project.Project, loadedManifest *manifest.Manifest) error {
+	for _, p := range projects {
+		for envName, env := range p.Configs {
+			for _, conf := range env {
+				switch conf[0].Type.(type) {
+				case config.ClassicApiType:
+					if loadedManifest.Environments[envName].Auth.Token == nil {
+						return fmt.Errorf("API: %s requires token", conf[0].Type)
+					}
+				default:
+					if loadedManifest.Environments[envName].Auth.OAuth == nil {
+						return fmt.Errorf("API: %s oatuh provided", conf[0].Type)
+					}
+				}
+			}
+		}
+	}
```

Comment:
pls check whether you can use the `project::ForEveryConfigDo(action)` helper here :)